java/src/main/java/com/company/automation/fundamentals/optional_task/optional_task_01/OptionalTask01.java

```diff
@@ -0,0 +1,414 @@
+        System.out.println("\nThe first number that contains min amount of different digits: " +
+                numberWithMinAmountOfDifferentDigits + "\n");
+
+        int quantityOfNumbersWithOnlyEvenDigits;
+        quantityOfNumbersWithOnlyEvenDigits = getQuantityOfIntegersWithOnlyEvenDigits(stringOfNumbers);
+        System.out.println("Quantity of numbers which contain only even digits: " +
+                quantityOfNumbersWithOnlyEvenDigits + "\n");
+
+        int quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits;
+        quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits =
+                getQuantityOfIntegersWithEqualAmountOfEvenAndUnevenDigits(stringOfNumbers);
+        System.out.println("Quantity of numbers which contain equal amount of even and uneven digits: " +
+                quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits + "\n");
+
+        int integerWithRisingDigitsSequence;
+        integerWithRisingDigitsSequence = getIntegerWithRisingDigitsSequence(stringOfNumbers);
+        System.out.println("The first number with rising digits sequence: " + integerWithRisingDigitsSequence + "\n");
+
+        int integerWithOnlyDifferentDigits;
+        integerWithOnlyDifferentDigits = getIntegerWithOnlyDifferentDigits(stringOfNumbers);
+        System.out.println("The first number with only different digits: " + integerWithOnlyDifferentDigits + "\n");
+
+    }
+
+
+    private static String inputIntegersAsAString() {
+        boolean hasWrongCharacters;
+        boolean hasDigits;
+        String line;
+        String message;
+        Scanner in = new Scanner(System.in);
+
+        message = "Enter integer numbers using SPACE to divide them:>> ";
+        do {
+            System.out.print(message);
+            Pattern restrictedCharacters = Pattern.compile("[^0-9-\\s+]");
+            Pattern mustHave = Pattern.compile("[\\d]");
+            line = in.nextLine();
+            Matcher matcher = restrictedCharacters.matcher(line);
+            hasWrongCharacters = matcher.find();
+            matcher = mustHave.matcher(line);
+            hasDigits = matcher.find();
+        } while (hasWrongCharacters || !hasDigits);
+
+        line = line.replaceAll("\\+\\s*", " ");
+        line = line.replaceAll("-\\s*", " -");
+        line = line.trim();
+        return line;
+    }
+
+
+    private static void printTheShortestNumber(String[] numbers) {
+        int minLength;
+        minLength = numbers[0].length();
+
+        for (String item : numbers) {
+            if (item.length() < minLength) {
+                minLength = item.length();
+            }
+        }
+        System.out.print("The shortest number(s) is(are): ");
+
+        for (String val : numbers) {
+            if (val.length() == minLength) {
+                System.out.print(val + " ");
+            }
+        }
+        System.out.print("(length: " + minLength + ")\n");
+    }
```

Comment:
The same as https://github.com/PavelYudzin/test-automation-training/pull/2/files#r525384197

Reply:
Rejected